editor/js/outcomes.js

```diff
@@ -281,21 +309,61 @@ ord.outcomes.unloadRaw = function (node, raws) {
   }
   if ($("input[value='upload']", node).is(':checked')) {
     const bytesValue = ord.uploads.unload(node);
-    raw.setBytesValue(bytesValue);
+    if (!isEmptyMessage(bytesValue)) {
+      raw.setBytesValue(bytesValue);
+    }
   }
   if ($("input[value='url']", node).is(':checked')) {
     const url = $('.outcome_raw_text', node).text();
-    raw.setUrl(url);
+    if (!isEmptyMessage(url)) {
+      raw.setUrl(url);
+    }
+  }
+  if (!isEmptyMessage(name) || !isEmptyMessage(raw)) {
+    raws.set(name, raw);
   }
-  raws.set(name, raw);
 };
 
 ord.outcomes.add = function () {
-  return addSlowly('#outcome_template', '#outcomes');
+  const node = addSlowly('#outcome_template', '#outcomes');
+  handler = function () {ord.outcomes.validateOutcome(node)};
+  addChangeHandler(node, handler);
+  return node;
 };
 
 ord.outcomes.addAnalysis = function (node) {
-  return addSlowly('#outcome_analysis_template', $('.outcome_analyses', node));
+  const analysisNode = addSlowly('#outcome_analysis_template', $('.outcome_analyses', node));
+  
+  // Handle name changes.
+  const nameNode = $('.outcome_analysis_name', analysisNode)
```

Comment:
Since the test only looks for round-trip consistency after loading/unloading (at least that is my impression), we aren't testing whether name changes are handled properly. However, this code is related to changing analysis fields to `analysis1` and `analysis2` in the `full.pbtxt` test